src/content/learn/fetching-data.md

```diff
@@ -0,0 +1,206 @@
+
+<Illustration>
+Illustration of the parallel work while getting dry cleaning
+</Illustration>
+
+Now, imagine you are trying to render the component `LATemperature`. The first thing you do is kick off the asynchronous work of reading the temperature file and you've received `tempPromise` as a Promise. Now, you're free to do the next set of work. But wait, the next instruction requires you to know the temperature, yet all we have is a Promise. The Promise may or may not be ready with the actual temperature value.
+
+Here is where `await` comes in. `await` is a keyword that is used before a Promise and tells JavaScript to pause running the current function and wait for the Promise to resolve.Then, when it is resolved, await will unwrap the Promise container to return the value. We use await on tempPromise to pause our component rendering until the file is read and receive the value returned.
+
+In our analogy, this would be similar to waiting at the dry cleaners for them to finish because the next chore on your todo list requires those clothes. Thus, preventing you from doing anything else in the meantime.
+
+Our other examples of fetching data from a database server or making a network request to an API endpoint are also asynchronous operations that return Promises.
+
+
+<DeepDive>
+
+####  When to use await {/*when-to-use-await*/}
+
+Generally, you'll want to kick off asynchronous work as early as possible so that you can maximize the amount of work you can do before needing to `await` the Promise returned.
+
+Due to the simple nature of our component, our example needed to immediately await the asynchronous operation because the component render was blocked on that data.
+
+Here is an example where we are able to kick off the asynchronous work and then do some synchronous work in the meantime, and then finally await the data just before it is used.
+
+```js
+async function HoveringButton({ userContext, label, lastTouchPoint }) {
+  // async operation
+  const localePromise = getLocale(userContext);
+
+  // expensive calculation
+  const offset = calculateNewOffset(lastTouchPoint);
+
+  return ...
+}
+```
+
+</DeepDive>
+
+You'll notice that our component function declarations now have an `async` prefix. This is also a JavaScript keyword and any function that uses `await` will need to be marked with `async`. If a function `await`s a Promise, it itself becomes asynchronous.
+
+Imagine being the parent function calling the `LATemperature` component function. When you call `LATemperature` and it gets to the instruction of `await`-ing `readFilePromise`, JavaScript execution returns to you and you receive a Promise that promises to give you the completed render of `LATemperature`.
```

Comment:
LATemperature is a component, not a normal function, so you wouldn't normally call it. Maybe the chaining aspect of promises should be explained with a non-component function: for example, you could create a function `readTemperature` that reads the file and then parses it and extracts the temperature as a number.